File: src/main/java/quebec/virtualite/backend/services/rest/WheelDTO.java
package quebec.virtualite.backend.services.rest;

import lombok.With;
import quebec.virtualite.backend.services.domain.entities.WheelEntity;

import javax.validation.constraints.NotBlank;

public record WheelDTO(
    @NotBlank
    @With
    String brand,

    @NotBlank
    @With
    String name
)
{
    public static WheelDTO toWheelDTO(WheelEntity entity)
    {
        return new WheelDTO(
            entity.getBrand(),
            entity.getName());
    }

    public WheelEntity toEntity(long id)
    {
        return new WheelEntity(id, brand, name);
    }
}

File: src/main/java/quebec/virtualite/backend/services/rest/WheelDTO.kt
package quebec.virtualite.backend.services.rest

import quebec.virtualite.backend.services.domain.entities.WheelEntity
import javax.validation.constraints.NotBlank

data class WheelDTO(
    @field:NotBlank
    val brand: String?,

    @field:NotBlank
    val name: String?
)
{
    constructor(entity: WheelEntity) : this(entity.brand, entity.name)

    fun toEntity(id: Long): WheelEntity = WheelEntity(id, brand!!, name!!)
}
